Quadriga/src/main/java/edu/asu/spring/quadriga/web/search/TextConceptSearchController.java

```diff
@@ -43,95 +40,83 @@
-                List<ITextFile> texts = new ArrayList<ITextFile>();
-                List<String> handles = new ArrayList<String>();
-
-                for (CreationEvent event : eventList) {
-                    String sourceRef = event.getSourceReference();
-
-                    // if we haven't seen the reference yet
-                    if (references.add(sourceRef)) {
-                        ITextFile txtFile = textFileManager.getTextFileByUri(sourceRef);
-                        if (txtFile == null || txtFile.getAccessibility() == ETextAccessibility.PRIVATE) {
-                            handles.add(sourceRef);
-                        } else {
-                            if (txtFile.getAccessibility() == ETextAccessibility.PUBLIC) {
-                                texts.add(txtFile);
-                                textFileManager.loadFile(txtFile);
-                                txtFile.setSnippetLength(40);
-                            }
-                        }
-                    }
-                }
-
-                model.addAttribute("references", handles);
-                model.addAttribute("texts", texts);
-
-                List<IProject> projects = projectManager.getProjectListByAccessibility(EProjectAccessibility.PUBLIC);
-
-                List<String> projectIds = new ArrayList<String>();
-                projects.forEach(p -> projectIds.add(p.getProjectId()));
-
-                ITransformedNetwork transformedNetwork = transformationManager
-                        .getSearchTransformedNetworkMultipleProjects(projectIds, conceptUri, INetworkStatus.APPROVED);
-
-                String json = null;
-                if (transformedNetwork != null) {
-                    json = jsonCreator.getJson(transformedNetwork.getNodes(), transformedNetwork.getLinks());
+        }
+        
+        for (String sourceRef : references) {
+            ITextFile txtFile = textFileManager.getTextFileByUri(sourceRef);
+            if (txtFile == null || txtFile.getAccessibility() == ETextAccessibility.PRIVATE) {
+                handles.add(sourceRef);
+            } else {
+                if (txtFile.getAccessibility() == ETextAccessibility.PUBLIC) {
+                    texts.add(txtFile);
+                    textFileManager.loadFile(txtFile);
+                    txtFile.setSnippetLength(40);
                 }
+            }
+        }
 
-                if (transformedNetwork == null || transformedNetwork.getNodes().size() == 0) {
-                    model.addAttribute("isNetworkEmpty", true);
-                }
+        ITransformedNetwork transformedNetwork = null;
+        if (conceptUriSearchList.size() >= 1 && eventIds.size() >= 1) {
+            List<INetworkNodeInfo> networkNodeList = networkManager.getNetworkNodes(eventIds);
+            transformedNetwork = transformationManager.getTransformedNetworkUsingNetworkNodesAndConcepts(networkNodeList, conceptUriSearchList);
```

Comment:
you're still redownloading all the info that you should have gotten in your first call to qstore, don't your?

Reply:
I don't think this re-downloads the info. The Neo4J returns the statement Ids that contain the the two searched concepts.  Now we want to create a network out of this statement id information. So we first fetch information about the nodes that belong to the previously fetched statement ids (from MySQL) and then construct the network using the node information.

Reply:
I think the transformationmanager makes a call to QStore. Please check the code.

Reply:
Yes. You are right. Checked the code. The transformationmanager calls QStore.